rust/src/lib.rs

```diff
@@ -0,0 +1,329 @@
+use js_sys::TypeError;
+use secp256k1::{Message, PublicKey, Secp256k1, SecretKey, Signature};
+use secp256k1_sys as ffi;
+use secp256k1_sys::CPtr;
+use wasm_bindgen::prelude::*;
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[macro_use]
+extern crate lazy_static;
+lazy_static! {
+    static ref SECP: Secp256k1<secp256k1::All> = Secp256k1::new();
+}
+
+const ZERO: [u8; 32] = [0u8; 32];
+
+enum Error {
+    BadPrivate,
+    BadPoint,
+    BadTweak,
+    BadHash,
+    BadSignature,
+    BadExtraData,
+}
+
+impl Error {
+    fn as_str(&self) -> &str {
+        match *self {
+            Error::BadPrivate => "Expected Private",
+            Error::BadPoint => "Expected Point",
+            Error::BadTweak => "Expected Tweak",
+            Error::BadHash => "Expected Hash",
+            Error::BadSignature => "Expected Signature",
+            Error::BadExtraData => "Expected Extra Data (32 bytes)",
+        }
+    }
+}
+
+impl From<Error> for JsValue {
+    fn from(error: Error) -> Self {
+        JsValue::from(TypeError::new(error.as_str()))
+    }
+}
+
+// Dummy Node.js Buffer type
+// TODO: Replace appropriate type. It might be good to subdivide into `Point`, `Scaler`, etc.
+type JsBuffer = Box<[u8]>;
```

Comment:
Duplicates with Buffer

Reply:
If I replace All instances of JsBuffer with Buffer, I get tons of errors.
So while they are conceptually duplicates. They are not interchangeable.

If you know a way to make them interchangeable please let me know.